### PR TITLE
support count aggregation

### PR DESCRIPTION
**Pull request by ylwu-amzn**
*Issue #, if available:*

*Description of changes:*

* support count aggregation

## Show more fields if choose count aggregation
Screen Shot 2020-05-20 at 9 53 06 AM

## Show only number field if choose average/sum/max/min/
Screen Shot 2020-05-20 at 9 55 34 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
Just prototype this feature. Will discuss with UX designer later.

File: public/pages/EditFeatures/utils/constants.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export enum SAVE_FEATURE_OPTIONS {
 
 export const AGGREGATION_TYPES = [
   { value: 'avg', text: 'average()' },
+  { value: 'value_count', text: 'count()' },
```
Comment (ohltyler) [inline public/pages/EditFeatures/utils/constants.ts:36]: small: is there a reason why this is `value_count` and not `count`?

File: public/pages/EditFeatures/utils/helpers.ts [hunk 2]
```diff
@@ -38,6 +41,26 @@ export const getNumberFields = (allFields: { [key: string]: string[] }) =>
...
+  const countableDataTypes = [
+    DATA_TYPES.NUMBER,
+    DATA_TYPES.BOOLEAN,
+    DATA_TYPES.KEYWORD,
+    DATA_TYPES.DATE,
+  ];
+  return getFieldOptions(
+    allFields,
+    Object.keys(allFields)
+      .map(filed => filed as DATA_TYPES)
```
Comment (ohltyler) [inline public/pages/EditFeatures/utils/helpers.ts:59]: small: typo in `field`

File: public/pages/EditFeatures/components/AggregationSelector/AggregationSelector.tsx [hunk 3]
```diff
@@ -50,9 +96,15 @@ export const AggregationSelector = (props: AggregationSelectorProps) => {
...
-              options={numberFields}
+              options={
+                get(form, `values.featureList.${props.index}.aggregationBy`) ===
+                'value_count'
+                  ? countableFields
+                  : numberFields
+              }
               {...field}
               onClick={() => {
+                console.log('form', form);
```
Comment (ohltyler) [inline public/pages/EditFeatures/components/AggregationSelector/AggregationSelector.tsx:107]: small: remove log here

File: public/pages/EditFeatures/utils/constants.ts (hunk 1 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/utils/constants.ts:36]: Check this https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-valuecount-aggregation.html

File: public/pages/EditFeatures/utils/helpers.ts (hunk 2 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/utils/helpers.ts:59]: sure, will fix

File: public/pages/EditFeatures/components/AggregationSelector/AggregationSelector.tsx (hunk 3 above)
Comment (ylwu-amzn) [reply public/pages/EditFeatures/components/AggregationSelector/AggregationSelector.tsx]: done

File: public/pages/EditFeatures/utils/constants.ts (hunk 1 above)
Comment (ohltyler) [reply public/pages/EditFeatures/utils/constants.ts:36]: Makes sense, thanks.

**Comment by ylwu-amzn** [comment]
> LGTM. Will approve it after you confirm with UX.

Confirmed with UX , this is good.

**Pull request merged**